### PR TITLE
fix: transform into classes with var fields

### PR DESCRIPTION
**Pull request by hughsimpson**
After creating https://github.com/scalalandio/chimney/issues/400 it struck me that the fix was probably actually pretty simple

### CODE REVIEW

File: chimney/src/test/scala/io/scalaland/chimney/TotalTransformerMutableFieldSpec.scala [hunk 1]
```diff
@@ -0,0 +1,100 @@
+package io.scalaland.chimney
+
+import scala.annotation.unused
+
+class TotalTransformerMutableFieldSpec extends ChimneySpec {
```
Comment (hughsimpson) [inline chimney/src/test/scala/io/scalaland/chimney/TotalTransformerMutableFieldSpec.scala:5]: This class is basically just a copy of `TotalTransformerDslsSeparationSpec` but with `var` fields on all Foo/Bar case classes. There may be a better place for the tests to go, but I'm not sufficiently familiar with the structure of the project... Nonetheless, these tests do pass without the accompanying fix

Comment (MateuszKubuszok) [reply chimney/src/test/scala/io/scalaland/chimney/TotalTransformerMutableFieldSpec.scala:5]: I'd say that for this issue adding a new entry in `IssuesSpec` would be enough. Just use enough transformations to cover all edge cases (transformInto/into.transform + maybe ditto for partials). If you want you can use either one test (`test("fix issue #400")`) or separate tests but grouped (`group("fix issue #400") { /* tests */ }`).

**Comment by MateuszKubuszok** [comment]
Hey, great thanks for the reproduction and fix! I enabled the tests and all of them pass, so I would happyly merge it somewhere later this week. Congratulations on your first contribution!

File: chimney/src/test/scala/io/scalaland/chimney/TotalTransformerMutableFieldSpec.scala (hunk 1 above)
Comment (hughsimpson) [reply chimney/src/test/scala/io/scalaland/chimney/TotalTransformerMutableFieldSpec.scala:5]: Ah great that sounds much neater, I'll rework the tests a little today then. Thanks for your quick response!

Comment (hughsimpson) [reply chimney/src/test/scala/io/scalaland/chimney/TotalTransformerMutableFieldSpec.scala]: Pushed neater tests 👍

**Comment by MateuszKubuszok** [comment]
> @MateuszKubuszok pushed up some neater tests if you can retrigger the CI 🙏

Done!

In general LGTM, the only reason I am not merging this PR (and others) is that GH Actions are eager to break on our benchmarking environment and I would like to have a benchmark for 0.8.0 succeed to finish and get published before I merge new commits. But if that would take too long to fix I would skip it.

**Review by MateuszKubuszok — APPROVED** [review]

**Comment by MateuszKubuszok** [comment]
> Oh yeah just looked. No idea what's going on there tbh, looks on the surface like something running out of memory for only 10kb but that obviously can't really be it unless there's a lot of those 10kb things

That's out own runner that we have to maintain ourselves. (For comparable benchmarks we have to have the same environment). It has some issues that we need to resolve.

**Comment by MateuszKubuszok** [comment]
Merged, I am not sure when I'll release it, as there are a few more fixes I want to add to master, but it surely will be a part of 0.8.1. Thank you again for your contribution!
